### DISCUSSION TITLE
WebCam Media (Live Audio+video) integration using C++

### DISCUSSION THREAD

**Discussion by KemsT** [paullouisageneau/libdatachannel#582] [Q&A]
Hello

I started to work on a native standalone c++ based video-conferencing application where I go through some [examples](https://github.com/paullouisageneau/libdatachannel/tree/master/examples) as a reference,  where I got such information like the client (Data Channels), media (media reception from C++), streamer (media sending from C++), and a signaling server where specifically media streaming is either predefined media (like example/streamer) or some got from java+html based code (like example/media).

I would like to  **make one standalone c++ based application that got live audio-video media data from the webcam and sends it to localhost.** 

So, 
Would you like to help me for capturing live media data through c++ based application?

**Comment by paullouisageneau** [comment]
This is a bit out of the scope of libdatachannel. For live audio or video you can use gstreamer or ffmpeg for instance.

There are basically two possiblities:

- The simplest way (very handy for prototyping) is to run the media capture, encoding and packetization pipeline as a separate process, send the RTP packets to a local socket and forward them in a track (like the media example but for sending). For instance, with gstreamer, the external pipeline will look like this on Linux for video (you may have to adapt it):
```
gst-launch-1.0 v4l2src device=/dev/video0 ! videoconvert ! x264enc tune=zerolatency bitrate=3000 speed-preset=superfast ! video/x-h264, profile=baseline ! rtph264pay pt=96 ! udpsink host=127.0.0.1 port=5000
```

- The proper way is to do the capture and encoding by calling the equivalent libraries from C++ and setting a equivalent pipeline. You can then let libdatachannel do the packetization in a similar way as in the streamer example.

**Comment by KemsT** [comment]
Thank you @paullouisageneau for clarifying things like MediaStream API is not supported by libdatachannel stack.

As of now, you know where I am stuck so **would you like to help me to give a suggestion of any c++ API which really good to me for capturing live audio/video data?**

Your suggestion is a great help for me.

Thank you...
